src/include/OpenImageIO/typedesc.h

```diff
@@ -241,14 +241,14 @@ struct OIIO_API TypeDesc {
     /// inequality is differing vector semantics.
     friend bool equivalent (const TypeDesc &a, const TypeDesc &b) {
         return a.basetype == b.basetype && a.aggregate == b.aggregate &&
-               (a.arraylen == b.arraylen || (a.arraylen == -1 && b.arraylen > 0)
-                                         || (a.arraylen > 0   && b.arraylen == -1));
+               (a.arraylen == b.arraylen || (a.arraylen < 0 && b.arraylen > 0)
```

Comment:
If I am reading this correctly it means that something like `float[3]` is equivalent to a `float[]` which happens to have size 4? However a `float[3]` is not equivalent to a `float[4]`?

Don't you want to `abs(a.arraylen) == abs(b.arraylen)` instead?

(I see the previous code also had this behavior, so maybe I'm not understanding the intent behind this method)